src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java

```diff
@@ -243,5 +258,15 @@ private SimpleQuantity toSimpleQuantity(final BigDecimal value) {
             simpleQuantity.setValue(value);
             return simpleQuantity;
         }
+
+        private String getSystemValue(final CodingType codingType) {
+            if (codingType == CodingType.SNOMED_CT_CODE) {
+                return SNOMED_CODING_SYSTEM;
+            } else if (codingType == CodingType.READ_CODE) {
+                return READ_CODING_SYSTEM;
+            }
+
+            return null;
```

Comment:
Looks like another map to me
```
private static final Map<CodingType, String> CODING_TYPE_SYSTEMS = Map.of(
    CodingType.SNOMED_CT_CODE, "http://snomed.info/sct",
    CodingType.READ_CODE, "http://read.info/readv2"
);
// ...
private String getSystemValue(final CodingType codingType) {
    return CODING_TYPE_SYSTEMS.getOrDefault(codingType, null);
```

Reply:
The problem is that the coding type can be null
e.g. With coding type: `INV+MQ+42T..:911::Serum vitamin B12'`
       Without coding type: `INV+MQ+:::B12/FOLATE'`

And the map would throw a null pointer exception. We could do a null check, but I think it'd make this private method less readable.

Reply:
Is it appropriate to be calling `.setSystem(null)`? My feeling is that we've tended to avoid calling setters on hapifhir objects when we don't have a value to set. If `codingType` can be null, then it should be returned as an `Optional` from `LaboratoryInvestigationResult`

Reply:
I agree with @andrewwa-kainos

Reply:
Makes sense, done 👍